app/src/main/java/com/depromeet/team6/presentation/ui/home/HomeViewModel.kt

```
@@ -691,6 +708,70 @@ class HomeViewModel @Inject constructor(
+
+    private enum class VersionResult {
+        UPDATE_REQUIRED,
+        UPDATE_OPTIONAL,
+        UP_TO_DATE
+    }
+
+    private fun getCurrentVersionName(): String? {
+        return try {
+            val packageInfo = context.packageManager.getPackageInfo(context.packageName, 0)
+            packageInfo.versionName
+        } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)